### PR TITLE
Fix compile with address sanitizer

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

sizeof is different in such case

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: lib/Containers/FlatHashMap.h [hunk 1]
```diff
@@ -53,8 +53,12 @@ template<class K, class V,
          class Eq = typename absl::flat_hash_map<K, V, Hash>::key_equal,
          class Allocator =
              typename absl::flat_hash_map<K, V, Hash, Eq>::allocator_type,
```
Comment (jsteemann) [inline lib/Containers/FlatHashMap.h:55]: I think the trailing comma needs to be moved into the `#if` here in order to compile.
Github doesn't allow me to make a code suggestion in its web UI here, unfortunately.

File: lib/Containers/FlatHashSet.h [hunk 2]
```diff
@@ -50,8 +50,12 @@ template<class T, class Hash = typename absl::flat_hash_set<T>::hasher,
          class Eq = typename absl::flat_hash_set<T, Hash>::key_equal,
          class Allocator =
              typename absl::flat_hash_set<T, Hash, Eq>::allocator_type,
```
Comment (jsteemann) [inline lib/Containers/FlatHashSet.h:52]: same here

File: lib/Containers/FlatHashMap.h (hunk 1 above)
Comment (MBkkt) [reply lib/Containers/FlatHashMap.h:55]: True

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by KVS85** [comment]
- [ ] https://jenkins01.arangodb.biz/job/arangodb-ANY-linux-san.x86-64/2104/

**Pull request merged**